File: Beecrowd-Problems/Rust/done!/problem1012.rs
use std::io;

fn read_and_parse_input() -> (f64, f64, f64) {
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    let values: Vec<&str> = input.trim().split(' ').collect();

    let a: f64 = match values[0].parse::<f64>() {
        Ok(num) => num,
        Err(_) => {
            eprintln!("Error while parsing value.");
            -1f64
        }
    };

    let b: f64 = match values[1].parse::<f64>() {
        Ok(num) => num,
        Err(_) => {
            eprintln!("Error while parsing value.");
            -1f64
        }
    };

    let c: f64 = match values[2].parse::<f64>() {
        Ok(num) => num,
        Err(_) => {
            eprintln!("Error while parsing value.");
            -1f64
        }
    };

    (a, b, c)
}

fn main() {

    let pi: f64 = 3.14159;
    let (a, b, c) = read_and_parse_input();

    let triangle_area: f64 = (a * c) / (2f64);
    let circle_area: f64 = pi * c * c;
    let trapezium_area: f64 = ((a + b) * c) / (2f64);
    let square_area: f64 = b * b;
    let rectangle_area: f64 = a * b;

    println!("TRIANGULO: {:.3}", triangle_area);
    println!("CIRCULO: {:.3}", circle_area);
    println!("TRAPEZIO: {:.3}", trapezium_area);
    println!("QUADRADO: {:.3}", square_area);
    println!("RETANGULO: {:.3}", rectangle_area);
}

File: Beecrowd-Problems/C#/done!/problem1012.cs
using System;
using System.Globalization;

namespace Program {
    class Program {

        private static Tuple<double, double, double> ReadAndParseInput() {
            string line = Console.ReadLine();
            var values = line.Split(' ');
            double a = double.Parse(values[0], CultureInfo.InvariantCulture);
            double b = double.Parse(values[1], CultureInfo.InvariantCulture);
            double c = double.Parse(values[2], CultureInfo.InvariantCulture);
            return Tuple.Create(a, b, c);
        }

        static void Main(string[] args) {

            CultureInfo.CurrentCulture = CultureInfo.InvariantCulture;
            CultureInfo.CurrentUICulture = CultureInfo.InvariantCulture;

            var valuesTuple = ReadAndParseInput();
            double a = valuesTuple.Item1;
            double b = valuesTuple.Item2;
            double c = valuesTuple.Item3;

            double triangleArea = a * c / 2;
            double circleArea = c * c * 3.14159;
            double trapeziumArea = (a + b) * c / 2;
            double squareArea = b * b;
            double rectangleArea = a * b;

            Console.WriteLine($"TRIANGULO: {triangleArea:F3}");
            Console.WriteLine($"CIRCULO: {circleArea:F3}");
            Console.WriteLine($"TRAPEZIO: {trapeziumArea:F3}");
            Console.WriteLine($"QUADRADO: {squareArea:F3}");
            Console.WriteLine($"RETANGULO: {rectangleArea:F3}");
        }
    }
}